charts/jxboot-helmfile-resources/templates/_helpers.tpl

```diff
@@ -1,14 +1,32 @@
 {{- define "ingressAnnotations" }}
 {{- $annotations := dict }}
+{{- $componentSpec := index .Values .component }}
+{{- if hasKey $componentSpec.ingress "annotations" }}
+{{- $_ := merge $annotations $componentSpec.ingress.annotations }}
+{{- end }}
 {{- $_ := merge $annotations .Values.ingress.annotations .Values.jxRequirements.ingress.annotations  }}
 {{- if not (hasKey $annotations "kubernetes.io/ingress.class") }}
+{{- $customIngressClass := "" }}
+{{- if $componentSpec.ingress.customIngressClass }}
+{{- $customIngressClass := $componentSpec.ingress.customIngressClass }}
+{{- else if hasKey .Values.ingress "customIngressClass" }}
+
+{{- if eq .component "docker-registry" }}
+{{- if hasKey .Values.ingress.customIngressClass "dockerRegistry" }}
+{{- $customIngressClass := index .Values.ingress.customIngressClass "dockerRegistry" }}
+{{- end }}
```

Comment:
It looks to me like you are missing an else here. As the PR is written .Values.ingress.customIngressClass is only used for docker-registry.
```suggestion
{{- end }}
{{- else }}
```

Reply:
yep - good catch... thanks

switched it for `{{- else if }}`